src/python/WMCore/Services/Rucio/Rucio.py

```diff
@@ -757,18 +757,21 @@ def pickRSE(self, rseExpression='rse_type=TAPE\cms_type=test', rseAttribute='ddm
         rsesWeight = []
 
         for rse in matchingRSEs:
-            attrs = self.cli.list_rse_attributes(rse)
+            rseAttrs = self.cli.list_rse_attributes(rse)
             if rseAttribute:
                 try:
-                    quota = float(attrs.get(rseAttribute, 0))
+                    attrValue = float(rseAttrs.get(rseAttribute, 0))
                 except (TypeError, KeyError):
-                    quota = 0
+                    attrValue = 0
             else:
-                quota = 1
-            requiresApproval = attrs.get('requires_approval', False)
-            if quota > minNeeded:
+                attrValue = 1
+            requiresApproval = rseAttrs.get('requires_approval', False)
+            if rseAttribute == "ddm_quota" and attrValue > minNeeded:
```

Comment:
Why are still we using `ddm_quota`? At the moment, we're providing `dm_weight` for every tape and disk RSE and we're setting `ddm_quota` the same was `dm_weight` just in case there's anyone using it. Since, we'll deprecate it in the future, I think it's better to drop it from MSOutput.

Reply:
It looks like with "ddm_quota" we were able to check two things:
* whether the storage had enough free space for the income data
* and get a weighted random RSE for that.

If we move everything to "dm_weight" now, my understand is that we need to carry those 2 checks separately. Which I believe would be:
* check the space available in a given RSE for a given Rucio account
* get a weighted random RSE amongh those that can take this data.

Of course, this would only be applicable for RSE specific data placement. Many places in WM are actually using a generic Disk expression, so it should be easy to adapt it over there.

If this answer/implementation is not really straight forward, I think it's better to deal with this "migration" in its own ticket. @hassan11196 please correct me and/or add anything that I might be missing. Thanks

Reply:
Hi @amaltaro we don't update `ddm_quota` attribute for tapes anymore. In fact, it has never been updated. It was hardcoded. I'm not really sure whether we should check whether there's enough space at a tape for a given dataset. It sounds over-protection to me. In principle, if a tape is running low in terms of free space, its weight will be low, so MSOutput will not likely choose it. If really need it, we can provide the remaining space at tape RSE in a different RSE attribute or tell you how this can be obtained simply with the python client [1] Let me know what you think. `dm_weight` can be used safely as a weight for all production disk and tape RSEs.

[1]
```
from rucio.client import Client
client = Client()
rse = "T1_IT_CNAF_Tape"
rse_usage = list(client.get_rse_usage(rse))

for metric in rse_usage:
    if metric["source"] == "static": static = metric["used"]
    if metric["source"] == "rucio": rucio = metric["used"]

free = (static - rucio)/1e15
print(f"Free space at {rse} is {free} PB")
```

Reply:
I agree with you. I am actually not concerned with Tape, starting with the assumption that we will always have enough space and select the tape RSE as a function of the `dm_weight` attribute.

My concern goes for Disk RSEs though. How can we be sure that a given RSE can receive input or pileup data? Do we need the 2 steps check I mentioned above? Or can we simply rely on the `dm_weight` attribute as well?

Reply:
Does MSOutput ever choose disk RSEs? It uses generic RSE expressions and it's Rucio's job to choose a destination which has enough space for the dataset, no?

Reply:
Not MSOutput, as it does a generic Disk tape data placement.
However, MSTransferor and MSPileup (input data placement) would require some storage availability check, IMO.